src/VirtoCommerce.CatalogModule.Data/Validation/ListEntriesMoveRequestValidator.cs

```diff
@@ -0,0 +1,52 @@
+using System.Linq;
+using FluentValidation;
+using VirtoCommerce.CatalogModule.Core.Model;
+using VirtoCommerce.CatalogModule.Core.Model.ListEntry;
+using VirtoCommerce.CatalogModule.Core.Services;
+using VirtoCommerce.Platform.Core.Common;
+
+namespace VirtoCommerce.CatalogModule.Data.Validation
+{
+    public class ListEntriesMoveRequestValidator : AbstractValidator<ListEntriesMoveRequest>
+    {
+        private readonly ICategoryService _categoryService;
+
+        public ListEntriesMoveRequestValidator(ICategoryService categoryService)
+        {
+            _categoryService = categoryService;
+
+            RuleFor(x => x).CustomAsync(async (moveInfo, context, token) =>
+            {
+                if (string.IsNullOrEmpty(moveInfo.Category))
+                {
+                    return;
+                }
+
+                var targetCategory = (await _categoryService.GetByIdsAsync(new[] { moveInfo.Category }, CategoryResponseGroup.WithOutlines.ToString())).FirstOrDefault();
+
+                if (targetCategory == null)
+                {
+                    context.AddFailure($"Destination category does not exist.");
+                    return;
+                }
+
+                foreach (var movedCategory in moveInfo.ListEntries.Where(x => x.Type.EqualsInvariant(CategoryListEntry.TypeName)))
+                {
+                    var movedCategoryPath = string.Join("/", movedCategory.Outline);
+                    var targetCategoryPath = string.Join("/", targetCategory.Outlines.FirstOrDefault());
```

Comment:
Why you do not use the `targetCategory.Outline ` here?

Reply:
Done, though it increases the complexity of mocking, as this property is calculated with recursion.